File: SortColors.java
/*
    Given an array nums with n objects colored red, white, or blue, sort
    them in-place so that objects of the same color are adjacent, with the
    colors in the order red, white, and blue.
    We will use the integers 0, 1, and 2 to represent the color red, white,
    and blue, respectively.

    You must solve this problem without using the library's sort function.
**/
class Solution {
    private void bubbleSort(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            for (int j = 0; j < nums.length; j++) {
                if (i == j) {
                    continue;
                }
                if (nums[i] < nums[j]) {
                    swap(nums, i, j);
                }
            }
        }
    }
    private void swap(int[] nums, int index1, int index2) {
        int tmp = nums[index1];
        nums[index1] = nums[index2];
        nums[index2] = tmp;
    }

    public void sortColors(int[] nums) {
        bubbleSort(nums);
    }
}

File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        counter = Counter(nums)
        red = 0
        white = counter[0]
        blue = white + counter[1]
        t_white = white
        t_blue = blue

        ptr = 0
        while ptr < len(nums) and (red < t_white or white < t_blue or blue < len(nums)):
            cur = nums[ptr]
            should = None
            if cur == 0:
                if red == ptr:
                    red += 1
                    ptr += 1
                else:
                    nums[red], nums[ptr] = nums[ptr], nums[red]
                    while red < len(nums) and nums[red] == 0:
                        red += 1
            elif cur == 1:
                if white == ptr:
                    white += 1
                    ptr += 1
                else:
                    nums[white], nums[ptr] = nums[ptr], nums[white]
                    while white < len(nums) and nums[white] == 1:
                        white += 1
            else:
                if blue == ptr:
                    blue += 1
                    ptr += 1
                else:
                    nums[blue], nums[ptr] = nums[ptr], nums[blue]
                    while blue < len(nums) and nums[blue] == 2:
                        blue += 1
            
